node_modules/oae-principals/lib/libraries/members.js

```diff
@@ -0,0 +1,293 @@
+ */
+var list = module.exports.list = function(group, visibility, opts, callback) {
+    LibraryAPI.Index.list(PrincipalsConstants.library.MEMBERS_INDEX_NAME, group.id, visibility, opts, function(err, entries, nextToken) {
+        if (err) {
+            return callback(err);
+        }
+
+        var memberEntries = _.map(entries, function(entry) {
+            return {'id': entry.resourceId, 'role': entry.value};
+        });
+
+        return callback(null, memberEntries, nextToken);
+    });
+};
+
+
+///////////////////
+// REGISTRATIONS //
+///////////////////
+
+/*!
+ * Register a library indexer that can provide resources to reindex the members library
+ */
+LibraryAPI.Index.registerLibraryIndex(PrincipalsConstants.library.MEMBERS_INDEX_NAME, {
+    'pageResources': function(libraryId, start, limit, callback) {
+        AuthzAPI.getAuthzMembers(libraryId, start, limit, function(err, memberEntries) {
+            if (err) {
+                return callback(err);
+            }
+
+            PrincipalsDAO.getPrincipals(_.pluck(memberEntries, 'id'), ['principalId', 'tenantAlias', 'visibility', 'smallPictureUri'], function(err, members) {
+                if (err) {
+                    return callback(err);
+                }
+
+                var resources = _.map(memberEntries, function(memberEntry) {
+                    var member = members[memberEntry.id];
+                    var role = memberEntry.role;
+                    return {
+                        'rank': _getMembersLibraryRank(libraryId, member, role),
+                        'resource': member,
+                        'value': role
+                    };
+                });
+
+                return callback(null, resources);
+            });
+        });
+    }
+});
+
+/*!
+ * Register a library search for resource members
+ */
+LibraryAPI.Search.registerLibrarySearch('members-library', ['user', 'group'], {
+    'association': {
+        'name': AuthzConstants.search.MAPPING_RESOURCE_MEMBERSHIPS,
+        'field': 'direct_memberships'
+    }
+});
+
+
+////////////
+// EVENTS //
+////////////
+
+/**
+ * Update the given principal in the members libraries if applicable changes have been made that
+ * indicate their rank or visibility will need to change
+ *
+ * @param  {RestContext}    ctx                 The REST context of the user that performed the update
+ * @param  {User|Group}     updatedPrincipal    The updated copy of the principal
+ * @param  {User|Group}     oldPrincipal        The old copy of the principal before update
+ */
+var _updatePrincipalInMembersLibraries = function(ctx, updatedPrincipal, oldPrincipal) {
+    // If there is no change in state of their profile picture or visibility, we don't have to make
+    // any updates
+    if (!_xor(_hasProfilePicture(updatedPrincipal), _hasProfilePicture(oldPrincipal)) &&
```

Comment:
Not really, because `_hasProfilePicture` is truthy/falsey, not explicit boolean values. I should change that though rather than have `_xor`